### DISCUSSION TITLE
Add Player#nameTag(Component) and Player#nameTag() : Component

### DISCUSSION THREAD

**Discussion by metabrixkt** [PurpurMC/Purpur#432] [__Dismissed__] [answered]
## Is your feature request related to a problem?
It's not related to any _bug_ or anything like that, but there IS some problem.
Example: There's is a player with name `Steve`
1. Add team with `/team add some_team`
2. Add a prefix with `/team modify some_team prefix {"text":"example prefix ","color":"red"}`
3. Set team's color to any color with `/team modify some_team color aqua`
4. Add Steve to this team with `/team join some_team Steve`
5. Now, in the <tab> player list AND above the player entity you'll see this
![image](https://user-images.githubusercontent.com/64396538/123540754-10928b80-d75a-11eb-9de8-d8d779aa331b.png)

**So what's the problem?**
There is a `Player#playerListName(Component)`. But if do this:
```java
Player player = Bukkit.getPlayer("Steve");
player.playerListName(Component.join(Component.empty(),
	Component.text("example prefix ").color(NamedTextColor.RED),
	Component.text("Steve").color(NamedTextColor.AQUA)
));
```
in the <tab> player list you'll see the same result as with `/team`, but not above the player entity. Name above the player is still `player.getName()`.

## Describe the solution you'd like.
I'd like to see these two new methods in the `org.bukkit.entity.Player` class:
```java
public void nameTag(final @Nullable net.kyori.adventure.text.Component nameTag);

public @Nullable net.kyori.adventure.text.Component nameTag();
```
(name of these methods isn't necessary)

## Describe alternatives you've considered.
I tried using displayName(Component), playerListName(Component) and customName(Component) methods at the same time: - First one isn't used by notchian server, it's only used by plugins;
- Second one sets name in the <tab> player list only;
- Third one has no effect even with setCustomNameVisible(true).

**Comment by BillyGalbreath** [comment, marked as answer]
The label above a player's head is controlled client side using the scoreboard team. There is no way to set any components directly to it otherwise.
